README.md

```diff
@@ -226,6 +229,118 @@ The 'meta' node contains a child node for each of the child sections in the json
+| Define Realm           | Specify the Keycloak realm for authentication. Usually the realm is set to `insights`, which is the default value used by the extension. You only need to change this field if a different realm has been configured on your server. |
+| Label Name             | Select the label you want to assign the connection to                                                                                                                                                                                |
+
+![Edit Insights connection](https://github.com/KxSystems/kx-vscode/blob/main/img/edit-insights-conn-form.png?raw=true)
+
+## Connection Labels
+
+Connection Labels allow you to categorize and organize your connections by assigning them distinct names and colors, making it easier to manage and locate specific connections within the application.
+
+![Connection Tree With Labels](https://github.com/KxSystems/kx-vscode/blob/main/img/conn-labels-tree.png?raw=true)
+
+### Create New Label
+
+To create a Label, start by **editing** or **creating** a connection. At the **bottom of the form**, you'll see a **Create New Label** button.
+
+![Create New Label Button](https://github.com/KxSystems/kx-vscode/blob/main/img/create-new-label-btn.png?raw=true)
+
+Clicking this button will open a **dialog** where you can enter a **Label name** and choose a **Label color** for it. You can then create the Label by clicking 'Create' or cancel the process by clicking 'Cancel.'
+
+| Property    | Description                                               |
+| ----------- | --------------------------------------------------------- |
+| Label Name  | Enter a name for the label.                               |
+| Label color | Select the color in the list of colors for the new label. |
+
+![Create New Label](https://github.com/KxSystems/kx-vscode/blob/main/img/create-new-label-dialog.png?raw=true)
+
+### Add Label to a connection
+
+To add a Label to a connection, start by **editing** or **creating** a connection. At the **bottom of the form**, you'll see a **Label Name** dropdown to select a Label, select the Label and click in **Edit or Create Connection**.
+
+| Property   | Description                          |
+| ---------- | ------------------------------------ |
+| Label Name | Select Label from the list of Labels |
+
+![Select Label](https://github.com/KxSystems/kx-vscode/blob/main/img/conn-labels.png?raw=true)
+
+### Rename Label
+
+Right-click the label at Connection Tree and select **Rename label**.
+
+![Rename Label Opt](https://github.com/KxSystems/kx-vscode/blob/main/img/labels-rename-opt.png?raw=true)
+
+A prompt will be displayed at the top of the screen with the field to edit the name of the Label.
```

Comment:
A prompt is displayed at the top of the screen where you can edit the name of the Label.